src/settings.js

```diff
@@ -0,0 +1,249 @@
+  // http_post, http_bind_address, http_api_key, consoleLevel, httpLogFormat, enabled_plugins
+  if (
+    oldConfig.http_port !== newConfig.http_port
+    || oldConfig.http_bind_address !== newConfig.http_bind_address
+    || oldConfig.http_api_key !== newConfig.http_api_key
+    || oldConfig.consoleLevel !== newConfig.consoleLevel
+    || oldConfig.httpLogFormat !== newConfig.httpLogFormat
+    || (oldConfig.enabled_plugins && oldConfig.enabled_plugins.sort().join('#'))
+      !== (newConfig.enabled_plugins && newConfig.enabled_plugins.sort().join('#'))
+  ) {
+    throw new errors.NonRecoverableConfigChangeError();
+  }
+
+  // Checking for changes in seed and multisig fields
+  if (oldConfig.seeds) {
+    let shouldStop = false;
+    for (const [seedKey, oldSeed] of Object.entries(oldConfig.seeds)) {
+      // If the new config has changed a seed, we nee to stop the wallets
+      if (newConfig[seedKey] !== oldSeed) {
+        shouldStop = true;
+        break;
+      }
+      // If we had a multisig configuration we need to check that it is still relevant
+      // Adding a new multisig configuration will not trigger this
+      if (oldConfig.multisig && oldConfig.multisig[seedKey]) {
+        // Here we check in order that the newConfig has a multisig config for the same seedKey
+        // Then we check that all fields are the same
+        if (
+          !(newConfig.multisig && newConfig.multisig[seedKey])
+          || newConfig.multisig[seedKey].total
+            !== oldConfig.multisig[seedKey].total
+          || newConfig.multisig[seedKey].numSignatures
+            !== oldConfig.multisig[seedKey].numSignatures
+          || newConfig.multisig[seedKey].pubkeys.sort().join(';')
+            !== oldConfig.multisig[seedKey].pubkeys.sort().join(';')
+        ) {
+          shouldStop = true;
+          break;
+        }
+      }
+    }
+    if (shouldStop) {
+      await _stopAllWallets();
+    }
+  }
+  if (newConfig.seeds) {
+    for (const seed of Object.values(newConfig.seeds)) {
+      try {
+        const isValid = walletUtils.wordsValid(seed);
+        if (!isValid.valid) {
+          throw new errors.NonRecoverableConfigChangeError();
+        }
+      } catch (err) {
+        // If the method throws InvalidWords, we should stop the service
+        if (err instanceof hathorLibErrors.InvalidWords) {
+          throw new errors.NonRecoverableConfigChangeError();
+        }
+        // If another error is thrown, just bubble it up
+        throw err;
+      }
+    }
+  }
+
+  // Checking for changes in server and network
+  if (
+    oldConfig.server !== newConfig.server
+    || oldConfig.network !== newConfig.network
+  ) {
+    await _stopAllWallets();
+    // change network and server in active configuration
+    hathorLibConfig.setNetwork(newConfig.network);
+    hathorLibConfig.setServerUrl(newConfig.server);
+    return;
+  }
+
+  // Checking for changes in tokenUid, gapLimit and connectionTimeout
+  if (
```

Comment:
Should the previous suggestion on this file not be accepted: is there a reason why we shouldn't update simultaneously these properties?

For example, if the `serverUrl` and `gapLimit` properties were changed at the same call, only the `serverUrl` would be updated, and another call would be necessary to update the `gapLimit`.

Reply:
Well, not in this case, since server/network changes would stop all wallets and the next time a wallet is started it would get the new gap-limit/connectionTimeout/tokenUid.
So effectively it would apply the changes to the gapLimit/connectionTimeout/tokenUid